internal/pkg/cli/svc_deploy.go

```diff
@@ -474,19 +480,40 @@ func (o *deploySvcOpts) getTargetApp() (*config.Application, error) {
 	return o.targetApp, nil
 }
 
-type errManifestIncompatibleWithEnvironment struct {
+type errFeatureIncompatibleWithEnvironment struct {
+	ws             wsEnvironmentsLister
 	missingFeature string
 	envName        string
+	curVersion     string
 }
 
-func (e *errManifestIncompatibleWithEnvironment) Error() string {
-	return fmt.Sprintf("environment %q is not on a version that supports the %q feature", e.envName, template.FriendlyEnvFeatureName(e.missingFeature))
+func (e *errFeatureIncompatibleWithEnvironment) Error() string {
+	if e.curVersion == "" {
+		return fmt.Sprintf("environment %q is not on a version that supports the %q feature", e.envName, template.FriendlyEnvFeatureName(e.missingFeature))
+	}
+	return fmt.Sprintf("environment %q is on version %q which does not support the %q feature", e.envName, e.curVersion, template.FriendlyEnvFeatureName(e.missingFeature))
 }
 
 // RecommendActions returns recommended actions to be taken after the error.
 // Implements main.actionRecommender interface.
-func (e *errManifestIncompatibleWithEnvironment) RecommendActions() string {
-	return fmt.Sprintf("You can upgrade your environment template by running %s.\n", color.HighlightCode(fmt.Sprintf("copilot env deploy --name %s", e.envName)))
+func (e *errFeatureIncompatibleWithEnvironment) RecommendActions() string {
+	envs, _ := e.ws.ListEnvironments() // Best effort try to detect if env manifest exists.
+	for _, env := range envs {
+		if e.envName == env {
+			return fmt.Sprintf("You can upgrade your environment template by running %s.", color.HighlightCode(fmt.Sprintf("copilot env deploy --name %s", e.envName)))
```

Comment:
```suggestion
			return fmt.Sprintf("You can upgrade your %s environment template by running %s.", color.HighlightCode(fmt.Sprintf("copilot env deploy --name %s", e.envName, e.envName)))
```

How about adding an environment name in the prompt as well?

Reply:
Done!